tests/test_function.py

```diff
@@ -415,6 +415,31 @@ def test_nested_partial_binding(self, reset_db) -> None:
         res = t.select(t.pb1, t.pb2, t.pb3).collect()
         assert res[0] == {'pb1': 'a y c default', 'pb2': 'x y c default', 'pb3': 'x y z changed'}
 
+    @staticmethod
+    @pxt.udf
+    def crt_test_udf(a: int, b: int, c: int = 5) -> pxt.Array[pxt.Int]:
+        return np.ones((b, c)) * a
+
+    def test_conditional_return_type(self, reset_db: None) -> None:
+        f = self.crt_test_udf
+
+        @f.conditional_return_type
```

Comment:
Why not move this into the same scope as crt_test_udf?

Reply:
Doesn't work due to some Python weirdness involving the `@staticmethod` declaration. Maybe there's a better way to handle this but I think this is fine for now.